components/automate-deployment/pkg/client/restore.go

```diff
@@ -521,6 +526,56 @@ func (r *DeploymentRestore) Restore(ctx context.Context) error {
 		return err
 	}
 
+	// We've already resolved/loaded the desired A2 manifest and used it to
+	// bootstrap habitat and deployment-service. Since the manifest that was
+	// persisted deployment-services backup may be different we'll set it to
+	// the version that we've resolved and that'll allow us to skip resolving
+	// it again in the restore gRPC call and having to persist it.
+	r.writer.Body("Updating service manifest")
+	if err := r.setManifest(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *DeploymentRestore) setManifest(ctx context.Context) error {
+	if r.a2Manifest == nil {
+		return status.New(
+			status.BackupRestoreError,
+			"failed to update to the desired A2 manifest",
+		)
+	}
+
+	js, err := json.Marshal(r.a2Manifest)
+	if err != nil {
+		return status.Wrap(err, status.MarshalError, "Failed to marshal package manifest to JSON")
+	}
+
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		deadline = time.Now().Add(5 * time.Second)
+	}
+
+	con, err := Connection(time.Until(deadline))
```

Comment:
it might be worth creating a function `ConnectionCtx(context.Context)` so you don't have to keep doing this deadline calculation

Reply:
There's a TODO to change Connection() to take a context, I was thinking I'd just do that there on another change e v e n t u a l l y